cmd/frontend/graphqlbackend/graphqlutil/connection_resolver.go

```diff
@@ -0,0 +1,271 @@
+package graphqlutil
+
+import (
+	"context"
+	"sync"
+
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+const DEFAULT_MAX_PAGE_SIZE = 100
+
+type ConnectionResolver[N ConnectionNode] struct {
+	store   ConnectionResolverStore[N]
+	args    *ConnectionResolverArgs
+	options *ConnectionResolverOptions
+	data    connectionData[N]
+	once    resolveOnce
+}
+
+type ConnectionNode interface {
+	ID() graphql.ID
+}
+
+type ConnectionResolverStore[N ConnectionNode] interface {
+	// ComputeTotal is required to fetch the total number of items.
+	ComputeTotal(context.Context) (*int32, error)
+	// ComputeTotal is required to fetch the list of nodes based on the pagination args.
+	ComputeNodes(context.Context, *database.PaginationArgs) ([]*N, error)
+	// MarshalCursor is used to generate cursor for a node. This is called for generating start and end cursors.
+	MarshalCursor(*N) (*string, error)
+	// UnMarshalCursor is used to get node id from the after/before cursors.
+	UnMarshalCursor(string) (*int, error)
```

Comment:
```suggestion
	UnmarshalCursor(string) (*int, error)
```

We should be consistent with the stdlib here. One example: https://pkg.go.dev/encoding/json#Unmarshal